src/source/src/connector_source.rs

```diff
@@ -261,19 +216,12 @@ impl ConnectorSource {
             }))
             .await?;
 
-        let streams = readers
-            .into_iter()
-            .map(inner_connector_source_reader_into_stream)
-            .collect::<Vec<_>>()
-            .merge()
-            .into_stream()
-            .boxed();
+        let stream = select_all(readers.into_iter().map(|r| r.into_stream())).boxed();
```

Comment:
Compared to `select_all`, `merge` will shuffle the streams before each polling. Not sure whether this matters.

Reply:
... which means that `select_all` has a potential starvation problem?

Reply:
I think so. 🥵

Reply:
`risingwave_common::util::select_all` 🤔

Reply:
Cool. 😄

Reply:
replaced by it

Reply:
Will we always prefer a fair `select_all`? :think:

Reply:
I think so. We tend to `select` on **unbounded** streams, where fairness should be critical.

Reply:
Then we can add a `disallowed-method` 😄

Reply:
I've added it in 07ee20a. But it didn't give any warning when I run `./risedev c`. Any problems here?

Reply:
Did you `cargo clean`?

Reply:
Nope. Why a clean is required? 😇

Reply:
Or is it a re-export?

Reply:
[Note that configuration changes will not apply for code that has already been compiled and cached](https://github.com/rust-lang/rust-clippy#:~:text=Note%20that%20configuration%20changes%20will%20not%20apply%20for%20code%20that%20has%20already%20been%20compiled%20and%20cached)

Reply:
No? Still have not managed to make it work.

Reply:
https://github.com/rust-lang/rust-clippy/issues/9558 Reported an issue to clippy 😇